discretize/utils/meshutils.py

```diff
@@ -255,3 +258,426 @@ def ExtractCoreMesh(xyzlim, mesh, meshType='tensor'):
         raise Exception("Not implemented!")
 
     return actind, meshCore
+
+
+def meshBuilderXYZ(
+    xyz, h,
+    padX=[0, 0], padY=[0, 0], padZ=[0, 0],
+    baseMesh=None,
+    expFact=1.3,
+    meshType='TENSOR',
+    verticalAlignment='top'
+):
+    """
+        Function to quickly generate a Tensor mesh
+        given a cloud of xyz points, finest core cell size
+        and padding distance.
+        If a baseMesh is provided, the core cells will be centered
+        on the underlaying mesh to reduce interpolation errors.
+
+        :param numpy.ndarray xyz: n x 3 array of locations [x, y, z]
+        :param numpy.ndarray h: 1 x 3 cell size for the core mesh
+        :param numpy.ndarray padDist: 2 x 3 padding distances [W,E,S,N,Down,Up]
+        [OPTIONAL]
+        :param numpy.ndarray padCore: Number of core cells around the xyz locs
+        :object SimPEG.Mesh: Base mesh used to shift the new mesh for overlap
+        :param float expFact: Expension factor for padding cells [1.3]
+        :param string meshType: Specify output mesh type: "TensorMesh"
+
+        RETURNS:
+        :object SimPEG.Mesh: Mesh object
+
+    """
+
+    assert meshType in ['TENSOR', 'TREE'], ('Revise meshType. Only ' +
+                                            ' TENSOR | TREE mesh ' +
+                                            'are implemented')
+
+    assert verticalAlignment in ['center', 'top'], ("verticalAlignment must be 'center' | 'top'")
+
+    # Get extent of points
+    limx = np.r_[xyz[:, 0].max(), xyz[:, 0].min()]
+    limy = np.r_[xyz[:, 1].max(), xyz[:, 1].min()]
+    limz = np.r_[xyz[:, 2].max(), xyz[:, 2].min()]
+
+    # Get center of the mesh
+    midX = np.mean(limx)
+    midY = np.mean(limy)
+
+    if verticalAlignment == 'center':
+        midZ = np.mean(limz)
+    else:
+        midZ = limz[0]
+
+    nCx = int(limx[0]-limx[1]) / h[0]
+    nCy = int(limy[0]-limy[1]) / h[1]
+    nCz = int(limz[0]-limz[1]+int(np.min(np.r_[nCx, nCy])/3)) / h[2]
+
+    if meshType == 'TENSOR':
+        # Make sure the core has odd number of cells for centereing
+        # on global mesh
+        if baseMesh is not None:
+            nCx += 1 - int(nCx % 2)
+            nCy += 1 - int(nCy % 2)
+            nCz += 1 - int(nCz % 2)
+
+        # Figure out padding cells from distance
+        def expand(dx, pad):
+            L = 0
+            nC = 0
+            while L < pad:
+                nC += 1
+                L = np.sum(dx * expFact**(np.asarray(range(nC))+1))
+
+            return nC
+
+        # Figure number of padding cells required to fill the space
+        npadEast = expand(h[0], padX[0])
```

Comment:
Naming again here and in many other places, e.g., `npad_east`, `pad_x`, `exp_fact`, ....